core_v2/sources/domains.move

```diff
@@ -1171,6 +1232,40 @@ module aptos_names_v2::domains {
         timestamp::now_seconds() >= expiration_time_sec
     }
 
+    /// Given signer, target address and transfer to address, return true if we should set target address and reverse lookup, false otherwise
+    public fun should_set_reverse_lookup_and_target_address_in_register_name(
+        signer_address: address,
+        target_address: Option<address>,
+        transfer_to_address: Option<address>,
+    ): bool {
+        let transfer_to_address = option::get_with_default(&transfer_to_address, signer_address);
+        let target_address = option::get_with_default(&target_address, signer_address);
+        if (target_address == signer_address && transfer_to_address == signer_address) {
+            true
+        } else {
+            false
+        }
+    }
+
+    /// Given signer, target address and transfer to address, return true if we should set target address, false otherwise
+    public fun should_set_target_address_in_register_name(
+        subdomain_name: Option<String>,
+        target_address: Option<address>,
+    ): bool {
+        if (!is_subdomain(subdomain_name)) {
+            // If signer is registering a domain, automatically set the name to point to target address, if not set use signer address
```

Comment:
I don't understand this case, can you elaborate?

Reply:
updated the comment, basically when registering a domain, we always set the target address of the new domain